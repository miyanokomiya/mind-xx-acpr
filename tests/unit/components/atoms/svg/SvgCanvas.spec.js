import { mount } from '@vue/test-utils'
import Target from '@/components/atoms/svg/SvgCanvas.vue'

describe('components/atoms/svg/SvgCanvas.vue', () => {
  describe('snapshot', () => {
    it('default', () => {
      const wrapper = mount(Target, {
        propsData: {
          x: 1,
          y: 2,
          width: 10,
          height: 20,
          scale: 100,
        },
      })
      expect(wrapper.element).toMatchSnapshot()
    })
    it('with selecting rectangle', () => {
      const wrapper = mount(Target, {
        propsData: {
          x: 1,
          y: 2,
          width: 10,
          height: 20,
          scale: 100,
        },
      })
      wrapper.setData({
        rectangleSelecting: true,
        downP: { x: 3, y: 4 },
        beforeMoveP: { x: 30, y: 40 },
      })
      expect(wrapper.element).toMatchSnapshot()
    })
  })
})
