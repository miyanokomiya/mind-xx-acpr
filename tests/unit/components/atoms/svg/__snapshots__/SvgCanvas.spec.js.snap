// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/atoms/svg/SvgCanvas.vue snapshot default 1`] = `
<div
  class="svg-canvas-wrapper"
  style="width: 10px; height: 20px;"
>
  <svg
    font-family="sans-serif"
    viewBox="1 2 0.1 0.2"
  >
     
    <!---->
  </svg>
</div>
`;

exports[`components/atoms/svg/SvgCanvas.vue snapshot with selecting rectangle 1`] = `
<div
  class="svg-canvas-wrapper"
  style="width: 10px; height: 20px;"
>
  <svg
    font-family="sans-serif"
    viewBox="1 2 0.1 0.2"
  >
     
    <path
      d="M 1,2 L 1.1,2 L 1.1,2.2 L 1,2.2 L 1,2 M 1.03,2.04 L 1.03,2.4 L 1.3,2.4 L 1.3,2.04"
      fill="gray"
      fill-opacity="0.4"
      stroke="none"
    />
     
    <rect
      fill="none"
      fill-opacity="1"
      height="0.3599999999999999"
      rx="3"
      ry="3"
      stroke="blue"
      stroke-dasharray="5"
      stroke-dashoffset="5"
      stroke-width="1"
      width="0.27"
      x="1.03"
      y="2.04"
    />
  </svg>
</div>
`;
